### ISSUE TITLE
Value added to a String field gets converted to an Integer

### ISSUE THREAD

**Issue by fedecarg** [strapi/strapi#1571]
**Informations**
- **Node.js version**: v10.1.0
- **npm version**: 5.6.0
- **Strapi version**: 3.0.0-alpha.12.6
- **Database**: MongoDB
- **Operating system**: Linux

**What is the current behavior?**

Strapi `content-manager` plugin converts large INT numbers with exponential notation. For example: 8370399913709911711799000 gets converted to 8.370399913709912e+24

This also happens regardless of the field type, Number (INT) or String.

**Steps to reproduce the problem**

- Add new item
- Add the value 8370399913709911711799000 to a String or Number (INT) field
- Save item
- Edit the item and the value should now be 8.370399913709912e+24

**What is the expected behavior?**

I expect a number or a string representation of a number without exponential notation

**Suggested solutions**

Seems to be a common problem with JS.

**Comment by lauriejim** [comment]
Hello @fedecarg , thank you for reporting this issue. We are currently on another feature and bug fix. Since time is lacking on our side, feel free investigate and submit a PR, we’ll appreciate your contribution on this issue! 

Check out the contributing guide to get started: https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md

**Comment by fedecarg** [comment]
I will consider this a bug rather than an enhancement. Because I I add a String field, I expect to get a String, and not an Integer.

**Comment by derrickmehaffy** [comment]
I can confirm this is a bug. Pasting it in an Integer field won't save on mysql due to the number being outside the maximum range. Pasting it into a string field results in the example.

**Comment by lauriejim** [comment]
Related to this issue https://github.com/strapi/strapi/issues/995

**Comment by lauriejim** [comment]
I just checked and it's the way MongoDB work.
Try to manually create a new document with this value `8370399913709911711799000`, it will be stored as `8.370399913709912e+24` value and it still a `Double` type. 
It's not stored as a `String`.

**Comment by animatedcreativity** [comment]
For me 1234567890 gets converted to 1234567890.0 even when field type is string. It keeps adding that .0 in the end. But, if I save A1234567890 then it will happily accept it as a string and will not add .0

Number checks should not be done on a value if its type is already string.
The same happens for text type also.

I changed the type to number and it got solved, but by data is a mix of characters and numbers sometimes. So, I am stuck.
